apis/Google.Cloud.BigQuery.V2/Google.Cloud.BigQuery.V2/Numeric.cs

```diff
@@ -452,14 +452,38 @@ public decimal ToDecimal(LossOfPrecisionHandling lossOfPrecisionHandling)
 
             // FIXME: Awful performance!
             string text = ToString();
```

Comment:
I was wondering if instead of doing truncation on this method, we could have `ToString(int maxDigits, LossOfPrecisionHandling lossOfPrecisionHandling)` and we would call that here `ToString(28, lossOfPrecisionHandling)`. If we get the string already truncated or a failure then we only need to parse to decimal. And I think it's easier to truncate when forming the string? I'll try to push a PR with that later.

But I'm fine with this as well.

Reply:
Frankly I get confused every time I look at this code, so whatever you can do would be good :)